src/System.Data.SqlClient/src/Resources/Strings.resx

```diff
@@ -1042,4 +1042,7 @@
   <data name="LocalDBNotSupported" xml:space="preserve">
     <value>LocalDB is not supported on this Platform.</value>
   </data>
+  <data name="PlatformNotSupported_DataSqlClient" xml:space="preserve">
+    <value>System.Data.SqlClient is not supported on this platform.</value>
+  </data>
```

Comment:
AFAIK AnyOS would be set when the configuration that gets picked doesn't  have OS defined, for example, in the list above, it would happen on `netstandard1.2`, `netstandard1.3` and `netstandard` configurations.